src/extractor/guidance/turn_analysis.cpp

```diff
@@ -1846,10 +1848,12 @@ std::vector<TurnCandidate> TurnAnalysis::mergeSegregatedRoads(
         const double correction_factor = (turn_candidates[1].angle) / 2;
         for (std::size_t i = 2; i < turn_candidates.size(); ++i)
             turn_candidates[i].angle += correction_factor;
-        turn_candidates[1].angle = 0;
+        turn_candidates[0] = merge(turn_candidates[0],turn_candidates[1]);
+        turn_candidates[0].angle = 0;
+        turn_candidates.erase(turn_candidates.begin() + 1);
     }
 
-    for (std::size_t index = 0; index < turn_candidates.size(); ++index)
+    for (std::size_t index = 2; index < turn_candidates.size(); ++index)
```

Comment:
Maybe just a comment on the loop as to why it starts at "2".  I can see it now reading the code, but it'd be nice if it was a bit more obvious.